src/components/CurrencyInputPanel/CurrencyInputPanel.container.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useCallback, useEffect, useMemo } from 'react'
+import debounce from 'lodash.debounce'
+
+import { CurrencyInputPanelComponent } from './CurrencyInputPanel.component'
+
+import {
+  useCurrencySearchModalSwap,
+  useCurrencySearchModalBridge,
+} from '../SearchModal/CurrencySearchModal/CurrencySearchModal.hooks'
+import { useBridgeInputValidation } from '../../pages/Bridge/ActionPanel/useBridgeInputValidation'
+
+import { CurrencyWrapperSource } from '../CurrencyLogo'
+import { CurrencyInputPanelProps } from './CurrencyInputPanel.types'
+
+import { CurrencySearchModalProvider } from '../SearchModal/CurrencySearchModal'
+import { normalizeInputValue } from '../../utils'
+
+export const CurrencyInputPanel = (currencyInputPanelProps: CurrencyInputPanelProps) => {
+  const searchModalContexts = useCurrencySearchModalSwap()
+
+  return (
+    <CurrencySearchModalProvider {...searchModalContexts}>
+      <CurrencyInputPanelComponent {...currencyInputPanelProps} />
+    </CurrencySearchModalProvider>
+  )
+}
+
+export const CurrencyInputPanelBridge = (currencyInputPanelProps: CurrencyInputPanelProps) => {
+  const searchModalContexts = useCurrencySearchModalBridge()
+
+  const {
+    value: valueRaw,
+    onUserInput: onUserInputRaw,
+    displayedValue,
+    setDisplayedValue,
+    disableCurrencySelect,
+  } = currencyInputPanelProps
+
+  const debounceOnUserInput = useMemo(() => {
+    return debounce(onUserInputRaw, 500)
+  }, [onUserInputRaw])
+
+  const onUserInput = useCallback(
+    (val: string) => {
+      const normalizedValue = normalizeInputValue(val)
+
+      setDisplayedValue && setDisplayedValue(normalizedValue)
+      debounceOnUserInput(normalizedValue)
+    },
+    [debounceOnUserInput, setDisplayedValue]
+  )
+
+  const value = useMemo(() => (disableCurrencySelect ? valueRaw : displayedValue ?? ''), [
+    disableCurrencySelect,
+    displayedValue,
+    valueRaw,
+  ])
```

Comment:
No need of useMemo here. You are just reading value here. No complex computation

Reply:
Fixed. You can take a look again :)